app/api/views/viewusers.py

```diff
@@ -0,0 +1,41 @@
+from flask import Flask, jsonify, Request, request, json, Response
+from flask.views import MethodView
+from app import app
+from app.api.models.users import User, users, userid
+
+
+class ViewUser(MethodView):
+
+    @app.route("/api/v1/users", methods=["GET"])
+    def all():
+        return jsonify(users)
+
+    @app.route("/api/v1/auth/signup", methods=["POST"])
+    def signup():
+        data = request.get_json()
+        firstname = data["firstname"]
+        lastname = data["lastname"]
+        email = data["email"]
+        password = data["password"]
+        user_data = {
+            "userid": userid,
+            "firstname": firstname,
+            "lastname": lastname,
+            "email": email,
+            "password": password
+        }
+        users.append(user_data)
+        return jsonify({"message": "Account was created successfully!"}), 201
+
+    @app.route('/api/v1/auth/login', methods=['POST'])
+    def login():
+        info = request.get_json()
+        email = info.get('email', None)
+        password = info.get('password', None)
+        if len(users) == 0:
+            return jsonify({'message': 'No users found'})
+        for user in users:
+            if user["email"] != email and user["password"] != password:
```

Comment:
So quick question, what validation are you looking for here?

Reply:
with the IF statement, I am checking to see if the list is empty. and if its I should get that response.
And the FOR loop with an IF statement, I am trying to crosscheck the email and password values if they do much the input data to the saved data

Reply:
I am asking about line 38 in particular

Reply:
Yes ! on Line 38, I am comparing user input values( email & password) at login with the user_data that was saved at signup. if they match, then the user should log in successfully otherwise they should be prevented to proceed and get a login failed response

Reply:
I see what you are trying to achieve but that line is not doing what you want it to, go back and test with different scenarios and see.

Reply:
> I see what you are trying to achieve but that line is not doing what you want it to, go back and test with different scenarios and see.

Okay. let me review it and see a better alternative